src/java.base/share/classes/jdk/internal/classfile/impl/DirectCodeBuilder.java

```diff
@@ -843,19 +974,23 @@ public CodeBuilder branch(Opcode op, Label target) {
 
     @Override
     public CodeBuilder nop() {
-        writeBytecode(NOP);
+        bytecodesBufWriter.writeU1(NOP);
         return this;
     }
 
     @Override
     public CodeBuilder aconst_null() {
-        writeBytecode(ACONST_NULL);
+        bytecodesBufWriter.writeU1(ACONST_NULL);
         return this;
     }
 
     @Override
     public CodeBuilder aload(int slot) {
-        writeLocalVar(BytecodeHelpers.aload(slot), slot);
+        if (slot >= 0 && slot <= 3) {
```

Comment:
Should we use `if ((slot & ~3) != 0)` for shorter bytecode?

Reply:
While fun, I wonder if such bit-fiddling optimizations obfuscate the code more than it helps performance. Would be good with some supporting evidence that it 1) helps interpreted performance and 2) that all JITs recognize and optimize this pattern well and without surprises.